gateway/handler_car.go

```diff
@@ -177,3 +176,37 @@ func rangeStrToByteRange(rangeStr string) (DagEntityByteRange, error) {
 		To:   &to,
 	}, nil
 }
+
+func getCarRootCid(imPath ImmutablePath) (cid.Cid, error) {
+	imPathStr := imPath.String()
+	if !strings.HasPrefix(imPathStr, "/ipfs/") {
+		return cid.Undef, fmt.Errorf("path does not have /ipfs/ prefix")
+	}
+
+	firstSegment, _, _ := strings.Cut(imPathStr[6:], "/")
+	rootCid, err := cid.Decode(firstSegment)
+	if err != nil {
+		return cid.Undef, err
+	}
+
+	return rootCid, err
+}
+
+func getCarEtag(r *http.Request, imPath ImmutablePath, params CarParams, rootCid cid.Cid) string {
```

Comment:
@hacdias @lidel did this get into a spec doc anywhere; or alternatively do you have tests that demonstrate what this output is going to be? I'm not entirely clear on what `ImmutablePath#String` does, but I think we end up with something like this from this function, would that be correct?

`"W/\"Qmfoobar.car." + xxhash("/ipfs/Qmfoobar/path/to/file.entity.100.400") + "\""`

Not strictly necessary to get this consistent between implementations I think, but if we end up with the same form then it might be quite useful. With strict ordering I believe we can ditch the `W/` and put in `.dfs` to the hash string.

Can you confirm whether `ImmutablePath#String` includes both the `/ipfs/` prefix and the CID? If so, could we leave that out and write the rest of this up into the path spec?

I recall having a discussion about this but can't find it so am not sure if it progressed!

Reply:
@rvagg afaik, the `Etag` is not going to be in the spec. It should be set (nor sure if it's a should, or a must), and as long as it's unique per request (path + arguments), it should be fine. We also don't have specs for other Etags.

Reply:
Here's my go at emulating this, adding `dups` too and removing the `W/` prefix: https://github.com/filecoin-project/lassie/pull/285 - could we encode this in the trustless and/or path spec? Even if just a recommendation and some fixtures?

Reply:
We had no guidance for Etags because old CAR responses were unspecified, no byte-for-byte guaranteee, Etag always weak.

 @rvagg  I agree this changes with IPIP-412 (which adds optional `dups` and `order`) – weak/strong Etag recommendation and some fixtures should be added there. 
 
 @hacdias will you have time to add it? if not I can eventually get to this before we finalize.